File: 30daysOFcode_TypeScript/day 13.ts
class Book {
    title: string;
    author: string;
    
    constructor(title: string, author: string) {
        if (this.constructor === Book) {
            throw new TypeError('Do not attempt to directly instantiate an abstract class.');
        }
        else {
            this.title = title;
            this.author = author;
        }
    }

    display(): void {
        console.log('Implement the \'display\' method!');
    }
}

class MyBook extends Book {
    price: number;

    constructor(title: string, author: string, price: number) {
        super(title, author);
        this.price = price;
    }

    display(): void {
        console.log('Title: ' + this.title);
        console.log('Author: ' + this.author);
        console.log('Price: ' + this.price);
    }
}

File: 30daysOFcode_Go/solution_set1/day 13.go
package main

import "fmt"

type Book struct {
	Title  string
	Author string
}

type Booker interface {
	Display()
}

type MyBook struct {
	Book   Book
	Price  int
}

func(b *MyBook) Display() {
	fmt.Printf("Title: %s\n", b.Book.Author)
	fmt.Printf("Author: %s\n", b.Book.Author)
	fmt.Printf("Price: %d\n", b.Price)
}

func main() {
	a := MyBook{Book{"The Alchemist","Paulo Coelho"},248}
	a.Display()
}